### PR TITLE
Chapter 7

### PR DESCRIPTION
**Pull request by sebing**
Chapter 7

### CODE REVIEW

File: chapter07/sxg/ex_7_5.pl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+use strict;
+
+my $find=0;
+my $line;
+
+print "I found:\n";
+while (<>) {
+	$line = $_;
+
+  	if (/[A-Z][a-z]+/) {
```
Comment (MarkWignall) [inline chapter07/sxg/ex_7_5.pl:12]: Does this + have an effect as there is no requirement for what is either side of the expression?

File: chapter07/sxg/ex_7_6.pl [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  	if (/wilma/) {
+  		$findW = 1;
+	}
+  	if (/fred/) {
+  		$findF = 1;
+	}
+	if($findF && $findW){
+		print "$line";	
+	}
```
Comment (MarkWignall) [inline chapter07/sxg/ex_7_6.pl:21]: This seems like a interesting way to do it but do $findF and $findW get returned to zero each loop? If not once fred and wilma have both appeared(even on different lines) would every line not be printed?

File: chapter07/sxg/ex_7_5.pl (hunk 1 above)
Comment (sebing) [inline chapter07/sxg/ex_7_5.pl:12]: First letter upper case [A-Z],
then an infinite number of(+) small cases [a-z]

This is what i thought the requirement ;)

File: chapter07/sxg/ex_7_6.pl (hunk 2 above)
Comment (pratyp) [inline chapter07/sxg/ex_7_6.pl:21]: Yeah, it would be useful to reset <code>findF</code> and <code>findW</code> at start/end of the loop.

Comment (qianliok) [inline chapter07/sxg/ex_7_6.pl:21]: That's a really good catch Mark!

**Comment by thatguy59** [comment]
Just in case you're not checking work e-mail (why would you want to anyway....), we're going to skip next week because a few of us are off, so we'll catch up the week after :wink:

**Pull request merged**